lib/lte_link_control/lte_lc.c

```diff
@@ -891,39 +891,61 @@ int lte_lc_psm_get(int *tau, int *active_time)
 	char active_time_str[9] = {0};
 	char tau_ext_str[9] = {0};
 	char tau_legacy_str[9] = {0};
+	char response[128] = { 0 };
+	const char ch = ',';
+	char *comma_ptr;
 
 	if ((tau == NULL) || (active_time == NULL)) {
 		return -EINVAL;
 	}
 
-	/* We expect match on at least two parameters, <Active-Time> and <Periodic-TAU-ext>.
-	 * Three matches will only happen on modem firmwares >= 1.2.0.
+	/* We need to parse the three last parameters, Active-Time, Periodic-TAU-ext and
+	 * Periodic-TAU. N.B. Periodic-TAU will not be present on modem firmwares < 1.2.0.
 	 */
-	err = nrf_modem_at_scanf("AT%XMONITOR",
-		"%%XMONITOR: "
-		"%*u,"		/* <reg_status> */
-		"%*[^,],"	/* <full_name> */
-		"%*[^,],"	/* <short_name> */
-		"%*[^,],"	/* <plmn> */
-		"%*[^,],"	/* <tac> */
-		"%*u,"		/* <AcT> */
-		"%*u,"		/* <band> */
-		"%*[^,],"	/* <cell_id> */
-		"%*u,"		/* <phys_cell_id> */
-		"%*u,"		/* <EARFCN> */
-		"%*u,"		/* <rsrp> */
-		"%*u,"		/* <snr> */
-		"%*[^,],"	/* <NW-provided_eDRX_value> */
-		"\"%8[0-1]\","	/* <Active-Time> */
-		"\"%8[0-1]\","	/* <Periodic-TAU-ext> */
-		"\"%8[0-1]\"",	/* <Periodic-TAU>, available only for modem fw >= v1.2.0 */
-		active_time_str, tau_ext_str, tau_legacy_str);
+
+	err = nrf_modem_at_cmd(response, sizeof(response), "AT%%XMONITOR");
 	if (err < 0) {
 		LOG_ERR("AT command failed, error: %d", err);
 		return -EFAULT;
-	} else if (err < 2) {
-		LOG_ERR("Active time and/or TAU interval is missing");
-		return -EBADMSG;
+	}
+
+	/* Skip over first 13 fields in AT cmd response by counting delimiters (commas). */
```

Comment:
I think it would be nice to have the format of the response here just to make it clear why we need to do this.
```
%XMONITOR: <reg_status>,[<full_name>,<short_name>,<plmn>,<tac>,<AcT>,<band>,<cell_id>,
<phys_cell_id>,<EARFCN>,<rsrp>,<snr>,<NW-provided_eDRX_value>,<Active-Time>,<Periodic-TAUext>,
<Periodic-TAU>]
```

Reply:
Sure, I added it now.